src/cryptoadvance/specter/templates/settings.jinja

```diff
@@ -3,35 +3,55 @@
-			<br><br>
-			Password:<br><input type="password" name="password" value="{{ password }}">
-			<br><br>
-			Host:<br><input type="text" name="host" type="text" value="{{ protocol }}://{{ host }}">
-			<br><br>
-			Port:<br><input type="text" name="port" type="text" value="{{ port }}">
-			<div class="note">
-				Default ports: <b>8332</b> for mainnet, <b>18332</b> for testnet, <b>18443</b> for Regtest, <b>38332</b> for signet
-			</div>
-			<br>
-			<h2>Miscellanous</h2>
-			<br>
-			Authentication:<br>
-			<select name="auth">
-				<option value="none" {% if auth=="none" %} selected="selected"{% endif %}>None</option>
-				<option value="rpcpasswordaspin" {% if auth=="rpcpasswordaspin" %} selected="selected"{% endif %}>RPC password as Pin</option>
-			</select>
-			<br><br>
+			{% if specter.config.auth == "none" or current_user.is_admin %}
+				<h2>Bitcoin JSON-RPC configuration</h2>
+				<br>
+				Username:<br><input type="text" name="username" value="{{ username }}">
+				<br><br>
+				Password:<br><input type="password" name="password" value="{{ password }}">
+				<br><br>
+				Host:<br><input type="text" name="host" type="text" value="{{ protocol }}://{{ host }}">
+				<br><br>
+				Port:<br><input type="text" name="port" type="text" value="{{ port }}">
+				<div class="note">
+					Default ports: <b>8332</b> for mainnet, <b>18332</b> for testnet, <b>18443</b> for Regtest, <b>38332</b> for signet
+				</div>
+				<br>
+				<h2>Miscellanous</h2>
+				<br>
+				Log Level:<br>
+				<select name="loglevel">
+					<option value="WARN" {% if loglevel=="WARN" %} selected="selected"{% endif %}>WARN</option>
+					<option value="INFO" {% if loglevel=="INFO" %} selected="selected"{% endif %}>INFO</option>
+					<option value="DEBUG" {% if loglevel=="DEBUG" %} selected="selected"{% endif %}>DEBUG</option>
+				</select>
+				<br><br>
+				Authentication:<br>
+				<select name="auth">
+					<option value="none" {% if auth=="none" %} selected="selected"{% endif %}>None</option>
+					<option value="rpcpasswordaspin" {% if auth=="rpcpasswordaspin" %} selected="selected"{% endif %}>RPC password as Pin</option>
+					<option value="usernamepassword" {% if auth=="usernamepassword" %} selected="selected"{% endif %}>Multiple Users</option>
+				</select>
+				<br><br>
+				{% if specter.config['auth'] == 'usernamepassword' %}
+					<button type="submit" class="btn" style="width: 100%;" name="action" value="adduser">Generate New Registration Link</button>
+					<br>
+					<span class="note">Default is username: admin, password: admin.</span><br>
+				{% endif %}
+			{% endif %}
+			{% if specter.config['auth'] == 'usernamepassword' %}
+				Specter Username:<br><input type="text" name="specter_username" type="text" value="{{ current_user.username }}"><br><br>
+				Specter Password:<br><input type="password" name="specter_password" type="text" placeholder="Set new password"><br>	
+				<br><br>			
```

Comment:
Let's add this part dynamically with javascript when the authentication method is changed. 
UX will be more intuitive in this case - the user selects `usernamepassword` setting and can set up admin login and password right away.
When authentication method is changed to `rpcpasswordaspin` or `none` - we remove this part of the form.